functions/api2.lua

```diff
@@ -927,22 +928,22 @@ function Details.UnitDamageSpellInfo (unitId, spellId, isLiteral, segment)
 		criticalHits = 0,
 		criticalDamage = 0,
 	}
-	
+
 	local playerObject = getActorObjectFromCombat (combatObject, 1, unitName)
 	if (not playerObject) then
 		return spellInfo
 	end
-	
+
 	local spellObject, spellId = getSpellObject (playerObject, spellId, isLiteral)
 	if (not spellObject) then
 		return spellInfo
 	end
 
+	local spellName = GetSpellInfo(spellId)

```

Comment:
Moving this out of the below check is unneded

Reply:
I actually do take this back, it would remove a nil for spell name in a lower check. Minor fix but a fix nonetheless